src/views/components/survey_components/question_time_setting.vue

```diff
@@ -0,0 +1,365 @@
+            <select v-model="selected">
+                <option v-for="question in questions">{{question.content}}</option>
+            </select>
+            &ensp;to &ensp;
+            <!-- The back up questions to choose from -->
+            <select v-model="unselected">
+                <option v-for="question in backup_questions">{{question.content}}</option>
+            </select>
+            <br/>
+            <br/>
+            <br/>
+            <button type="submit" v-on:click="replace(selected, unselected)">
+                Change</button>
+            &ensp;
+            <button type="submit" v-on:click="cancel">Cancel</button>
+            </br>
+        </div>
+        <br/>
+
+        <h2><b>Set the start time</b></h2>
+        <p>(Note: Please use the same format as default)</p>
+        <input type="datetime-local" v-model="start_time"/>
+        <h2><b>Set the end time</b></h2>
+        <p>(Note: Please use the same format as default)</p>
+        <input type="datetime-local" v-model="end_time"/>
+        <br/><br/><br/><br/>
+
+        <button type="submit" v-on:click="finish">
+            Finish the setting</button>
+    </div>
+</template>
+
+<script>
+export default {
+    name: "question_time_setting",
+    data: function () {
+        return {
+            survey_package: null,
+            num_locked: 0,
+            selected: null,
+            unselected: null,
+            survey_id: null,
+            name: null,
+            edit_click: false,
+            checkbox_indices: [],
+            question_id_array: [],
+            check_status: [0, 0],
+            term: "201801",
+            start_time: "2000-01-01T00:00:00",
+            end_time: "2000-01-01T00:00:00",
+            // The questions is gonna be a list of question object
+            questions: null,
+            // The backup questions is gonna be a list of back up questions
+            backup_questions: []
+        }
+    },
+    created: async function () {
+        this.survey_id = this.$route.query.survey_id;
+        this.name = this.$route.query.name;
+        // Fetch the list of questions from the API call
+        let fetched = await fetch(
+            "get_info.php?what=surveys&user_id=admin0&survey_id=1"
+        );
+        let fetchedJSON = await fetched.json();
+        // Parse the survey_package returned back into the data
+        if (fetchedJSON.TYPE === "error") {
+            // TODO: redirect to the 404 page
+            return 404;
+        } else if (fetchedJSON.TYPE === "survey_package") {
+            // assign value to the 'num_locked'
+            this.survey_package = fetchedJSON.DATA;
+            if (fetchedJSON.DATA.dept_number_locked != null) {
+                this.num_locked += parseInt(fetchedJSON.DATA.dept_number_locked);
+            }
+            if (fetchedJSON.DATA.course_number_locked != null) {
+                this.num_locked += parseInt(fetchedJSON.DATA.course_number_locked);
```

Comment:
What API are you basing this on? I don't recall `course_number_locked` in the api at all...It needs to be documented if it is part of the datastructure. Also, it should be names something like `number_locked_by_course` because right now it looks like "the course-number is locked" rather than "the number of things locked for this course"

Reply:
Daniel will make a change of the names on the API.